gammapy/makers/safe.py

```diff
@@ -281,7 +281,13 @@ def run(self, dataset, observation=None):
         dataset : `Dataset`
             Dataset with defined safe range mask.
         """
-        mask_safe = np.ones(dataset._geom.data_shape, dtype=bool)
+
+        if dataset.mask_safe:
+            mask_safe = dataset.mask_safe.data
+            if np.all(mask_safe == False):
```

Comment:
Maybe even remove this "fast-track" path? Of course the `SafeMaskMaker` does not modify the mask if it is all `False` from the beginning, however it maybe still adds some meta data in future. Performance wise it should be completely negligible...

Reply:
`MapDatasetOnOff.background` errors out if `counts_off` is None (because map arithmetic fails). To avoid this, I adapted map artithmetic such that `map X None = None`, where `X`   is any `operator`.
This is reasonable for multiplication or division, but `a + None = None` is strange (currently it just fails).

The other solution would be to manually set `background`, `npred_background` to `None` in `MapDatasetOnOff` if `counts_off` is None.

What would you suggest?

Reply:
Thinking about this, the behaviour I introduced in maps is not nice. I'm reverting it and rather adapting `MapDatasetOnOff.background`

Reply:
Yes, that's the preferable solution. If  `counts_off` is `None` than `npred_background` should return `None` as well. Arithmetics involving `None` should always fail!